### PR TITLE
Adding Copyright Linter

### PR DESCRIPTION
**Pull request by CursedRock17**
Meant to be part of a group of PRs that will replace #120, this pull request adds just the copyright linters and resolves the issues that come with it. This pull request is in preparation of issue #29.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -57,11 +57,11 @@ ament_export_targets(${PROJECT_NAME})
 ament_export_dependencies(rclcpp rcutils std_msgs)
 
 if(BUILD_TESTING)
-  find_package(ament_lint_auto REQUIRED)
+  find_package(ament_cmake_copyright)
```
Comment (ahcorde) [inline CMakeLists.txt:60]: Add the follwing to the package.xml
```xml
  <test_depend>ament_cmake_copyright</test_depend>
```

CI is failing https://build.ros2.org/job/Rpr__message_filters__ubuntu_noble_amd64/16/console

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: include/message_filters/subscriber.h [hunk 2]
```diff
@@ -152,8 +146,8 @@ struct message_type <M, true>
   using type = typename M::custom_type;
 };
 
-template <typename M> 
-struct message_type <M, false> 
+template <typename M>
+struct message_type <M, false>
```
Comment (clalancette) [inline include/message_filters/subscriber.h:150]: I'm going to suggest we not make this change here.  In other words, I think this PR should strictly be about copyright; leave these other fixes to other PRs when we introduce the code linters.

File: include/message_filters/subscriber.h [hunk 3]
```diff
@@ -162,7 +156,7 @@ template <typename M>
 using message_type_t = typename message_type<M>::type;
 
 template<class M, class NodeType = rclcpp::Node>
-class Subscriber 
+class Subscriber
```
Comment (clalancette) [inline include/message_filters/subscriber.h:159]: Same here.

File: include/message_filters/sync_policies/latest_time.h [hunk 4]
```diff
@@ -49,7 +43,7 @@ Synchronizer<latest_policy> sync_policies(latest_policy(), caminfo_sub, limage_s
 sync_policies.registerCallback(callback);
 \endverbatim
 
- * May also take an instance of a `rclcpp::Clock::SharedPtr` from `rclpp::Node::get_clock()` 
+ * May also take an instance of a `rclcpp::Clock::SharedPtr` from `rclpp::Node::get_clock()`
```
Comment (clalancette) [inline include/message_filters/sync_policies/latest_time.h:46]: Same here; let's put these in a different PR.

File: src/message_filters/__init__.py [hunk 5]
```diff
@@ -64,7 +64,7 @@ def signalMessage(self, *msg):
             cb(*(msg + args))
 
 class Subscriber(SimpleFilter):
-    
+
```
Comment (clalancette) [inline src/message_filters/__init__.py:67]: Same with this change; let's put it in a different PR.

File: src/message_filters/__init__.py [hunk 6]
```diff
@@ -183,7 +183,7 @@ def getOldestTime(self):
         if not self.cache_times:
             return None
         return self.cache_times[0]
-        
+
```
Comment (clalancette) [inline src/message_filters/__init__.py:186]: And same here.

File: test/test_approxsync.py [hunk 7]
```diff
@@ -1,63 +1,64 @@
...
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 # POSSIBILITY OF SUCH DAMAGE.
 
+import random
+
+import unittest
+
```
Comment (clalancette) [inline test/test_approxsync.py:32]: Let's make these changes in a different PR (same for all of the rest of the changes in this file).

File: test/test_message_filters_cache.py [hunk 8]
```diff
@@ -1,54 +1,51 @@
...
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 # POSSIBILITY OF SUCH DAMAGE.
 
+import unittest
+
```
Comment (clalancette) [inline test/test_message_filters_cache.py:31]: Let's make all of these changes to this file in a different PR.

File: test/test_message_traits.cpp [hunk 9]
```diff
@@ -1,36 +1,31 @@
...
-*  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
-*  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
-*  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
-*  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
-*  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
-*  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
-*  POSSIBILITY OF SUCH DAMAGE.
-*********************************************************************/
+// Copyright 2010, Willow Garage, Inc. All rights reserved.
+// Copyright 2024, Open Source Robotics Foundation, Inc. All rights reserved.
```
Comment (clalancette) [inline test/test_message_traits.cpp:2]: This one isn't correct; this should be:

```suggestion
// Copyright 2022, Kenji Brameld All rights reserved.
```

File: test/test_simple.cpp [hunk 10]
```diff
@@ -1,36 +1,31 @@
...
-*  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
-*  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
-*  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
-*  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
-*  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
-*  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
-*  POSSIBILITY OF SUCH DAMAGE.
-*********************************************************************/
+// Copyright 2010, Willow Garage, Inc. All rights reserved.
+// Copyright 2024, Open Source Robotics Foundation, Inc. All rights reserved.
```
Comment (clalancette) [inline test/test_simple.cpp:2]:
```suggestion
// Copyright 2010, Willow Garage, Inc. All rights reserved.
```

**Review by clalancette — CHANGES_REQUESTED** [review]

**Comment by CursedRock17** [comment]
The copyrighting should be all filled out, just a lot of finicky whitespace issues I had to resolve

**Review by ahcorde — APPROVED** [review]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/21109/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/15408/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/21823/)

**Comment by clalancette** [comment]
Fantastic, this looks good to me.  Thanks for splitting this out.

**Pull request merged**